src/Orleans.Core/Configuration/OptionLogger/IOptionFormatter.cs

```diff
@@ -32,6 +32,28 @@ public interface IOptionFormatterResolver<T>
         IOptionFormatter<T> Resolve(string name);
     }
 
+    public class OptionsFormatterComparer : IEqualityComparer<IOptionFormatter>
```

Comment:
Worried this will mask junk in our system.  A duplicate formatter indicates a bug right?

Reply:
current option formatter behavior doesn't guarantee just one formatter per option. If a user call ConfigureNamedOptions twice, it will print two identical ones. AFAIK, this is accepted by three people touched this piece of code, hence not a bug?

Reply:
In other areas we use `Properties` on either `ISiloHostBuilder` or `HostBuilderContext`.
Eg here: https://github.com/dotnet/orleans/blob/51fea8aedbec0ef7a33ce9a76692cf392ff507b1/src/Orleans.Runtime/Hosting/CoreHostingExtensions.cs#L25
or here: https://github.com/dotnet/orleans/blob/51fea8aedbec0ef7a33ce9a76692cf392ff507b1/src/Orleans.TestingHost.Legacy/LegacySiloBuilderConfigurator.cs#L27

Reply:
If you're worried that `HostBuilderContext` wont be accessible because you've only got `IServiceCollection` at the point where this is used, we can modify SiloHostBuilder to ensure that `HostBuilderContext`, etc, are always added to `IServiceCollection` before any of the other `ConfigureServices` delegates are run.

Reply:
I'm not sure I follow. This dedup is to remove duplicated OptionsFormatter if user configure the same OptionFormatter twice. Can you explain more on why is this related to the `Properties`?

Reply:
> Current option formatter behavior doesn't guarantee just one formatter per option.

Understood.  The system does not inforce this.  If we can figure out how to enforce this, I'd be for doing so.  If not, my preference would be to leave the misconfiguration observable rather than masking it.

I suspect we may be able to enforce the desired behavior using what Reuben suggested, but i've not looked into it.

Reply:
@xiazen, Jason was saying above "Worried this will mask junk in our system" because the PR merely filters duplicates after the fact rather than preventing them from being added in the first place.

What I'm suggesting is to prevent duplicates rather than filter them out when you're reading the formatters.

Ultimately it doesn't affect too much, it's just a matter of cleanliness.

Reply:
Oh I see. I can do a pass using `Properties`. I introduced here in this PR mainly because I run into the duplication problem in streaming. If I add ConfigureNamedOptions in IPersistentStreamProviderConfigurator.Configure<TOptions>, then it is really hard for user to figure out when to call ConfigureNamedOptions and when not, and the whole ConfigureNamedOptions and ConfigureFormatter is kind of fragile. 

But thanks @ReubenBond  I will take a pass on that.